media-sound/pulseaudio/pulseaudio-14.99.1.ebuild

```diff
@@ -0,0 +1,386 @@
+"
+# This is a PDEPEND to avoid a circular dep
+# TODO: Verify that alsa-plugins actually needs matching ${MULTILIB_USEDEP}
+PDEPEND="
+	alsa? ( alsa-plugin? ( >=media-plugins/alsa-plugins-1.0.27-r1[pulseaudio,${MULTILIB_USEDEP}] ) )
+"
+
+# alsa-utils dep is for the alsasound init.d script (see bug 155707); TODO: read it
+# NOTE: if (e)logind is now mandatory, then the act-group/audio is needed only with system-wide
+RDEPEND="${RDEPEND}
+	system-wide? (
+		alsa? ( media-sound/alsa-utils )
+		acct-user/pulse
+		acct-group/audio
+		acct-group/pulse-access
+	)
+"
+
+BDEPEND="
+	doc? ( app-doc/doxygen )
+	orc? ( >=dev-lang/orc-0.4.15 )
+	system-wide? ( dev-util/unifdef )
+	test? ( >=dev-libs/check-0.9.10 )
+	sys-devel/gettext
+	sys-devel/m4
+	virtual/libiconv
+	virtual/libintl
+	virtual/pkgconfig
+"
+
+DOCS=( NEWS README ) # todo is useless to install
+
+pkg_pretend() {
+	if use daemon; then
+		CONFIG_CHECK="~HIGH_RES_TIMERS"
+		WARNING_HIGH_RES_TIMERS="CONFIG_HIGH_RES_TIMERS:\tis not set (required for enabling timer-based scheduling in pulseaudio)\n"
+		check_extra_config
+
+		if linux_config_exists; then
+			local snd_hda_prealloc_size=$(linux_chkconfig_string SND_HDA_PREALLOC_SIZE)
+			if [[ -n "${snd_hda_prealloc_size}" && "${snd_hda_prealloc_size}" -lt 2048 ]]; then
+				elog "For best experience with HDA devices set the pre-allocated"
+				elog "sound buffer to at least 2048 kB. The current value is:"
+				elog "\tCONFIG_SND_HDA_PREALLOC_SIZE=${snd_hda_prealloc_size}"
+			fi
+		fi
+	fi
+}
+
+pkg_setup() {
+	linux-info_pkg_setup
+	gnome2_environment_reset # bug 543364 # TODO: read it
+}
+
+src_prepare() {
+	default
+
+	if [[ -n ${COMMIT} ]]; then
+		# This file really should be upstream's responsibility but what can you do other than hack
+		# together a work-around for an upstream's tarball generator lacking required integration?
+		if [[ -n ${_SNAPSHOT_FIX_GITVERSION} && ${_SNAPSHOT_FIX_GITVERSION} -ge 1 ]]; then
+			echo ${PV%_*}-${COMMIT:0:8} > .tarball-version
+		fi
+	fi
+}
+
+meson_native_true() {
+	if multilib_is_native_abi; then
+		echo "-D${1}=true"
+	else
+		echo "-D${1}=false"
+	fi
+}
+
+meson_native_feature() {
+	# Example usage: $(meson_native_feature foo bar) -> -Dbar=enabled if foo is true
+	# and the native ABI is being configured, disabled otherwise; if bar is omitted,
+	# then bar is replaced with foo e.g. $(meson_native_feature foo) -> -Dfoo=enabled.
+	echo "-D${2:-${1}}=$(multilib_native_usex ${1} enabled disabled)"
+}
+
+multilib_src_configure() {
+	local emesonargs=(
+		-Dadrian-aec=false # Not packaged?
+		--localstatedir="${EPREFIX}"/var
+		-Dmodlibexecdir="${EPREFIX}/usr/$(get_libdir)/${P}"
+#		-Dsystemduserunitdir=$(systemd_get_userunitdir)
+		-Dudevrulesdir="$(get_udevdir)"/rules.d
+		-Dbashcompletiondir="$(get_bashcompdir)" # Alternatively DEPEND on app-shells/bash-completion for pkg-config to provide the value
+		$(meson_native_feature alsa)
+		$(meson_native_true bluetooth bluez5)
```

Comment:
this should account for actual use flags via `multilib_native_usex` just like meson_native_feature does, and probably be named like `meson_native_use_enable`
as far as I can see currently it is just `true` for native ABI which is not what we want

Reply:
I know this's a long thread but there already was a [discussion on that](https://github.com/gentoo/gentoo/pull/20388#discussion_r613812151) right here as well as talks about the same topic regarding another package with a different Gentoo developer privately, and they unanimously did not approve of any kind of ebuild specific prefix to custom functions.

Reply:
`foo_boolean` is more than fine :) main point is that first branch in this helper should be something like this
```
foo_boolean() {
    local use_flag="${1}"
    local meson_option="${2:-${1}}"
    
    	if multilib_is_native_abi; then
    		echo "-D${meson_option}=$(multilib_native_usex ${use_flag} true false)"
    	else
    		echo "-D${meson_option}=false"
    	fi
}
```
i.e. not just return `true` but actually follow use flag

I'm not sure if it actually makes sense to disable meson options for non-native ABI this way, maybe only for disabling `daemon` for non-native ABI?
Initially I opted to employ `multilib_native_usex` to let non-native ABI follow use flags as well, so maybe there should be two helpers, `foo_native_abi_boolean` and `foo_boolean` ?

Reply:
Oh, I see what you mean, I think something may have gotten mixed up, when I ported those functions from the PipeWire jack-sdk & multilib PR. I'll think some more about this, before making any commits addressing it, but it looks at least the daemon one might, indeed, be behaving incorrectly. Thanks for noticing.

Reply:
Correct, just looked at pipewire ebuild - it is using `meson_native_enabled` function to unconditionally enable meson features for native abi, note those meson features of pipewire do not depend on use flags.

Reply:
Hi @pinkflames I see packages are now switching to `meson-multilib` eclass which provides `meson_native_use_bool` `meson_native_use_feature` and `meson_native_enabled` already, no need to define these in the ebuild. `pipewire` ebuild is already converted.

Reply:
Thanks for the heads up. I actually converted my personal pipewire fork to meson-multiib.eclass just days after it became available. The pulseaudio ebuild hasn't been converted yet but it's the next thing I'll do.